client/src/client.ts

```diff
@@ -1972,7 +1979,7 @@ class RenameFeature extends TextDocumentFeature<TextDocumentRegistrationOptions>
 		});
 	}
 
-	protected registerLanguageProvider(options: TextDocumentRegistrationOptions): Disposable {
+	protected registerLanguageProvider(options: RenameRegistrationOptions): Disposable {
```

Comment:
RenameRegistrationOptions should also be the type of the TextDocumentFeature (e.g. TextDocumentFeature<RenameRegistrationOptions>)